sql/core/src/main/scala/org/apache/spark/status/protobuf/sql/SQLExecutionUIDataSerializer.scala

```diff
@@ -64,7 +64,7 @@ class SQLExecutionUIDataSerializer extends ProtobufSerDe {
       getOptional(ui.hasCompletionTime, () => new Date(ui.getCompletionTime))
     val errorMessage = getOptional(ui.hasErrorMessage, () => ui.getErrorMessage)
     val metrics =
-      ui.getMetricsList.asScala.map(m => SQLPlanMetricSerializer.deserialize(m)).toSeq
+      ui.getMetricsList.asScala.map(m => SQLPlanMetricSerializer.deserialize(m))
```

Comment:
If no `toSeq`, one operation is about 10ns. After adding `toSeq`, the delay will increase linearly with the length of the original data: 

- 240ns when input length is 10
- 1740ns when input length is 100
- 16600ns when input length is 1000